contracts/examples/erc1155-transfer/FxERC1155ChildTunnel.sol

```diff
@@ -7,124 +7,140 @@ import { Create2 } from "../..//lib/Create2.sol";
-            }
-            
-            else {
-                revert("FxERC1155ChildTunnel: INVALID_SYNC_TYPE");
-            }
-      }
-      
-      function _mapToken(bytes memory syncData) internal returns (address) {
-          (address rootToken, string memory uri) = abi.decode(syncData , (address, string));
-          
-          address childToken = rootToChildToken[rootToken];
-          require(childToken == address(0x0), "FxERC1155ChildTunnel: ALREADY_MAPPED");
-          
-          bytes32 salt = keccak256(abi.encodePacked(rootToken));
-          childToken = createClone(salt, tokenTemplate);
-          IFxERC1155(childToken).initialize(address(this),rootToken, string(abi.encodePacked(uri)));
-          
-          rootToChildToken[rootToken] = childToken;
-          emit TokenMapped(rootToken, childToken);
-
-          return childToken;
-      }
-      
-      function _syncDeposit(bytes memory syncData) internal {
-          (address rootToken, address depositor, address user, uint256 id, uint256 amount, bytes memory data) = abi.decode(syncData, (address, address, address, uint256, uint256, bytes));
-          
-          address childToken = rootToChildToken[rootToken];
-          IFxERC1155 childTokenContract = IFxERC1155(childToken);
-          
-          childTokenContract.mint(user, id, amount, data);
-          
-      }
-      
-      function _syncDepositBatch(bytes memory syncData) internal {
-          (address rootToken, address depositor, address user, uint256[] memory ids, uint256[] memory amounts, bytes memory data) = abi.decode(syncData, (address, address, address, uint256[], uint256[], bytes));
-          
-          address childToken = rootToChildToken[rootToken];
-          IFxERC1155 childTokenContract = IFxERC1155(childToken);
-          
-          childTokenContract.mintBatch(user, ids, amounts, data);
-    
-      }
-      
-     
-      function _isContract(address _addr) private view returns (bool){
-        uint32 size;
-        assembly {
-            size := extcodesize(_addr)
-        }
-        return (size > 0);
-    }
+	bytes32 public constant DEPOSIT = keccak256("DEPOSIT");
+	bytes32 public constant DEPOSIT_BATCH = keccak256("DEPOSIT_BATCH");
+	bytes32 public constant WITHDRAW = keccak256("WITHDRAW");
+	bytes32 public constant WITHDRAW_BATCH = keccak256("WITHDRAW_BATCH");
+	bytes32 public constant MAP_TOKEN = keccak256("MAP_TOKEN");
+	//string public constant URI = "FXERC1155URI" ;
 
+	event TokenMapped(address indexed rootToken, address indexed childToken);
+	// root to child token
+	mapping(address => address) public rootToChildToken;
+	address public tokenTemplate;
+
+	constructor(address _fxChild, address _tokenTemplate) FxBaseChildTunnel(_fxChild) {
+		tokenTemplate = _tokenTemplate;
+		require(_isContract(_tokenTemplate), "Token template is not contract");
+	}
+
+	function withdraw(address childToken, uint256 id, uint256 amount, bytes memory data) public {
```

Comment:
The changes are these 4 functions, and the internal `_withdraw` functions below. Had to reset the formatting here to be similar to all other contracts.